src/sqlfluff/cli/commands.py

```diff
@@ -491,7 +495,7 @@ def lint(
     extra_config_path: Optional[str] = None,
     ignore_local_config: bool = False,
     **kwargs,
-) -> NoReturn:
+) -> None:
```

Comment:
Unrelated change: The correct type hint is `None`. IIUC, `NoReturn` means the function never returns (i.e. always throws an exception). PyCharm was flagging this.